protocol/ops.go

```diff
@@ -0,0 +1,37 @@
+package protocol
+
+import (
+	"log"
+
+	"github.com/coniks-sys/coniks-go/merkletree"
+)
+
+func (d *ConiksDirectory) Register(uname string, key []byte) (*merkletree.TemporaryBinding, int) {
+	d.RLock()
+	// check whether the name already exists
+	// in the directory before we register
+	ap := d.pad.Lookup(uname)
```

Comment:
I think we need to be return ``ap`` as the proof of absence along with the TB and the error.